sw/device/tests/spi_host_macronix_flash_test.c

```diff
@@ -0,0 +1,115 @@
+
+static status_t pinmux_connect_spi_host1_to_bob(const dif_pinmux_t *pinmux) {
+  // CSB.
+  TRY(dif_pinmux_output_select(pinmux, kTopEarlgreyPinmuxMioOutIob7,
+                               kTopEarlgreyPinmuxOutselSpiHost1Csb));
+  // SCLK.
+  TRY(dif_pinmux_output_select(pinmux, kTopEarlgreyPinmuxMioOutIoa3,
+                               kTopEarlgreyPinmuxOutselSpiHost1Sck));
+  // SD0.
+  TRY(dif_pinmux_input_select(pinmux, kTopEarlgreyPinmuxPeripheralInSpiHost1Sd0,
+                              kTopEarlgreyPinmuxInselIoa5));
+  TRY(dif_pinmux_output_select(pinmux, kTopEarlgreyPinmuxMioOutIoa5,
+                               kTopEarlgreyPinmuxOutselSpiHost1Sd0));
+
+  // SD1.
+  TRY(dif_pinmux_input_select(pinmux, kTopEarlgreyPinmuxPeripheralInSpiHost1Sd1,
+                              kTopEarlgreyPinmuxInselIoa4));
+  TRY(dif_pinmux_output_select(pinmux, kTopEarlgreyPinmuxMioOutIoa4,
+                               kTopEarlgreyPinmuxOutselSpiHost1Sd1));
+  // SD2.
+  TRY(dif_pinmux_input_select(pinmux, kTopEarlgreyPinmuxPeripheralInSpiHost1Sd2,
+                              kTopEarlgreyPinmuxInselIoa8));
+  TRY(dif_pinmux_output_select(pinmux, kTopEarlgreyPinmuxMioOutIoa8,
+                               kTopEarlgreyPinmuxOutselSpiHost1Sd2));
+  // SD3.
+  TRY(dif_pinmux_input_select(pinmux, kTopEarlgreyPinmuxPeripheralInSpiHost1Sd3,
+                              kTopEarlgreyPinmuxInselIoa7));
+  TRY(dif_pinmux_output_select(pinmux, kTopEarlgreyPinmuxMioOutIoa7,
+                               kTopEarlgreyPinmuxOutselSpiHost1Sd3));
+  return OK_STATUS();
+}
+
+static void init_test(dif_spi_host_t *spi_host) {
+  dif_pinmux_t pinmux;
+  mmio_region_t base_addr =
+      mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR);
+  CHECK_DIF_OK(dif_pinmux_init(base_addr, &pinmux));
+  pinmux_connect_spi_host1_to_bob(&pinmux);
+
+  base_addr = mmio_region_from_addr(TOP_EARLGREY_SPI_HOST1_BASE_ADDR);
+  CHECK_DIF_OK(dif_spi_host_init(base_addr, spi_host));
+
+  CHECK(kClockFreqPeripheralHz <= UINT32_MAX,
+        "kClockFreqPeripheralHz must fit in uint32_t");
+
+  CHECK_DIF_OK(dif_spi_host_configure(spi_host,
+                                      (dif_spi_host_config_t){
+                                          .spi_clock = 1000000,
+                                          .peripheral_clock_freq_hz =
+                                              (uint32_t)kClockFreqPeripheralHz,
+                                      }),
+               "SPI_HOST config failed!");
+
+  CHECK_DIF_OK(dif_spi_host_output_set_enabled(spi_host, true));
+}
+
+bool test_main(void) {
+  dif_spi_host_t spi_host;
+
+  init_test(&spi_host);
+  enum MacronixVendorSpecific {
+    kDeviceId = 0x1B20,
+    kManufactureId = 0xC2,
+    kPageQuadProgramOpcode = 0x38,
+    // The Macronix flash requires that the addr in sent using 4 lanes as the
+    // data when issuing the `kPageQuadProgramOpcode` operation.
+    kPageQuadProgramAddrWidth = 4,
```

Comment:
FYI, this style of SPI flash IO is known as 1-4-4 mode (1 lane for the opcode, 4 lanes for the address phase and 4 lanes for the data phase).  Support for this particular configuration should be described by the SFDP (and similarly, support for the 1-1-4 style used by the Winbond part should be described by its SFDP).

For the purpose of these tests, I don't think we need to be parsing the SFDPs, I just wanted to fill in a bit of background information.

The only thing you _might_ want to change is the comment to use the SPI terminology:
```
// The Macronix flash `4PP` opcode operates in 1-4-4 mode.
```

Reply:
Thanks for the information! I changed the comment.